### PR TITLE
Basic infra for the login in the backoffice

### PR DESCRIPTION
**Pull request by fpquintao**

### CODE REVIEW

File: icubam/backoffice/handlers/login.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  def post(self):
+    email = self.get_argument("email", "")
+    password = self.get_argument("password", "")
+    authenticate = self._authenticate(email, password)
+    if authenticate[0]:
+      # TODO(fpquintao): Set the cookie with the user id.
+      self.set_secure_cookie(self.COOKIE, tornado.escape.json_encode(email))
+      return self.redirect(self.get_argument("next", "/"))
+    # Failed to login, GET page again with the error message.
+    return self.redirect("/login" + self._makeErrorMessage(authenticate[1]))
```
Comment (olivierteboul) [inline icubam/backoffice/handlers/login.py]: maybe self.ROUTE ?

File: icubam/backoffice/handlers/home.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
 from icubam.backoffice.handlers.base import BaseBOHandler
 
+import tornado.web
 
 class HomeBOHandler(BaseBOHandler):
```
[the remark this answers is not in the document]
Comment (fpquintao) [reply icubam/backoffice/handlers/home.py]: done everywhere.

File: icubam/backoffice/handlers/login.py (hunk 1 above)
Comment (fpquintao) [reply icubam/backoffice/handlers/login.py]: ops, done.

**Comment by fpquintao** [comment]
Please take another look; I've addressed the comments & added tests.

File: icubam/backoffice/server_test.py [hunk 3]
```diff
@@ -17,5 +17,15 @@ def get_app(self):
...
   def test_homepage_without_cookie(self):
-    response = self.fetch(home.HomeBOHandler.ROUTE)
+    response = self.fetch(home.HomeHandler.ROUTE)
     self.assertEqual(response.code, 200)
+
+  def test_login(self):
+    response = self.fetch(login.LoginHandler.ROUTE)
+    self.assertEqual(response.code, 200)
+
+  def test_login_with_error(self):
```
Comment (olivierteboul) [inline icubam/backoffice/server_test.py]: nice!

**Pull request merged**